### DISCUSSION TITLE
Withdrawal, Consumption, and Demand variables

### DISCUSSION THREAD

**Discussion by BakedEggsAndCream** [iiasa/CWatM#119] [Q&A] [answered]
Hi all, 

I was wondering about the meaning of certain demand variables within the model. Specifically whether they represent consumption or withdrawal and potential rather than actual. For example, variables such as totalWaterDemand, industryDemand, irrDemand. I know some variables have more explicit names such as act_totalWaterWithdrawal.

I was also looking for potential consumption and potential withdrawal. I know ptotuse and ptotww outputs can be download for CWatM on the ISIMIP data repository, so those variables must exist somewhere in the model. There are variables that have names like: pot_domesticConsumption. Although these variables do not seem to have any units associated with them and whenever I attempted to output them I received an error message:
[error_message.txt](https://github.com/iiasa/CWatM/files/14878724/error_message.txt)

I didn't get this error for any other output, so I assume you can't output that variable.

**Comment by luanshilaogao** [comment]
Trying to use GPT to solve the problem you encountered is a good approach. The error shows that your two arrays are not in the correct form. Perhaps you can try printing two variables to see the comparison.
I have successfully output the result for the pot-domesticConsumption variable, perhaps there is a problem somewhere.
[pot_domesticConsumption_monthavg.zip](https://github.com/iiasa/CWatM/files/14930489/pot_domesticConsumption_monthavg.zip)

**Comment by mikhailsmilovic** [comment, marked as answer]
Some useful abstraction/withdrawal, consumption, and demand CWatM variables are included here. *All units are in metres*.

## Withdrawals (actual)

![MicrosoftTeams-image (36)](https://github.com/iiasa/CWatM/assets/20219171/91eea020-9fb1-4342-9fd5-4cfa30ddf5a9)
This figure shows the total abstractions from a 30-day simulation. All bolded output variables are in the figure—there are relatively zero surface water abstractions/withdrawals.

### _when sectorSourceAbstractionFractions = True_

**GW_Irrigation** 
**GW_Domestic**
**GW_Livestock** 
**GW_Industry**
 
**Res_Irrigation**
**Res_Domestic** 
**Res_Livestock** 
**Res_Industry**

**Lake_Irrigation**
**Lake_Domestic**
**Lake_Livestock** 
**Lake_Industry**

**Lift_Irrigation**
**Lift_Domestic**
**Lift_Livestock** 
**Lift_Industry**

**Channel_Irrigation** 
**Channel_Domestic** 
**Channel_Livestock** 
**Channel_Industry**

### _standard withdrawal variables_

**act_irrWithdrawal**
**act_nonIrrWithdrawal**
act_domWithdrawal 
act_indWithdrawal 
act_livWithdrawal
act_SurfaceWaterAbstract 
act_irrNonpaddyWithdrawal
nonFossilGroundwaterAbs 

pot_GroundwaterAbstract
_when limitAbstraction=False, this equals groundwater abstractions minus nonFossilGroundwaterAbs_ 

## Consumption (actual)
act_indConsumption
act_domConsumption 
act_livConsumption
act_nonpaddyConsumption
act_paddyConsumption
act_livConsumption
act_nonIrrConsumption
act_totalIrrConsumption 


## Demand
_≈potential consumption. When limitAbstract = False, then withdrawals = demand._

domesticDemand
industryDemand
livestockDemand
nonIrrDemand
totalIrrDemand
unmetDemand

**Reply by BakedEggsAndCream** [reply]
Thank you very much!
